app/src/lib/git/diff.ts

```diff
@@ -10,6 +10,72 @@ import { DiffType, IRawDiff, IDiff, IImageDiff, Image } from '../../models/diff'
 
 import { DiffParser } from '../diff-parser'
 
+function wrapAndParseDiff(args: string[], path: string, name: string, callback: (diff: IRawDiff) => Promise<IDiff>, successExitCodes?: Set<number>): Promise<IDiff> {
+
+  return new Promise<IDiff>((resolve, reject) => {
+    const commandName = `${name}: git ${args.join(' ')}`
+    log.debug(`Executing ${commandName}`)
+
+    const startTime = (performance && performance.now) ? performance.now() : null
+
+    const process = GitProcess.spawn(args, path)
+    process.stdout.setEncoding('binary')
+
+    const stdout = new Array<Buffer>()
+
+    process.stdout.on('data', (chunk) => {
+      if (chunk instanceof Buffer) {
+        stdout.push(chunk)
+      } else {
+        stdout.push(Buffer.from(chunk))
+      }
+    })
+
+    process.stdout.on('close', () => {
+      // process.on('exit') may fire before stdout has closed, so this is a
+      // more accurate point in time to measure that the command has completed
+      // as we cannot proceed without the contents of the stdout stream
+      if (startTime) {
+        const rawTime = performance.now() - startTime
+        if (rawTime > 1000) {
+          const timeInSeconds = (rawTime / 1000).toFixed(3)
+          log.info(`Executing ${commandName} (took ${timeInSeconds}s)`)
+        }
+      }
+
+      const maximumStringSize = 268435441
```

Comment:
Since this is a pretty important magic number, what do you think about pulling it up to file scope and documenting it?